pghoard/restore.py

```diff
@@ -788,8 +808,18 @@ def _write_status_output_to_file(self, output_file):
             }
         )
 
-    def _wait_for_jobs_to_complete(self):
+    def _wait_for_jobs_to_complete(self, pool):
         while self.jobs_in_progress():
+            with self.lock:
+                if self.jobs_to_retry:
+                    to_queue = []
+                    for item in self.data_files:
+                        if item.id in self.jobs_to_retry:
+                            self.pending_jobs.add(item.id)
+                            self.jobs_to_retry.remove(item.id)
+                            to_queue.append(item)
+                    for item in to_queue:
```

Comment:
This isn't outside of the lock now (also need to declare `to_queue` earlier to allow moving this few levels to left)

Reply:
Sorry I went too fast.